tests/end2end/playwright/requests-api.spec.js

```diff
@@ -0,0 +1,170 @@
+// @ts-check
+import {expect, test} from '@playwright/test';
+
+/**
+ * Playwright Page
+ * @typedef {import('@playwright/test').APIResponse} APIResponse
+ */
+
+const url = 'api.php/admin';
+
+/**
+ * Check for a JSON response about the metadata
+ * @param {APIResponse} response The response object
+ * @returns {Promise<any>} The JSON response
+ */
+export async function checkJson(response) {
+    expect(response.ok()).toBeTruthy();
+    expect(response.status()).toBe(200);
+    expect(response.headers()['content-type']).toBe('application/json');
+    return await response.json();
+}
+
+test.describe('Not connected',
+    {
+        tag: ['@requests', '@readonly'],
+    }, () => {
+
+        test('GET', async ({request}) => {
+            const response = await request.get(url + "/repositories", {});
+            expect(response.status()).toBe(401)
+        });
+
+        test('PUT', async ({request}) => {
+            const response = await request.put(url + "/repositories", {});
+            expect(response.status()).toBe(501)
+        });
+    }
+);
+
+test.describe('Connected',
+    {
+        tag: ['@requests', '@readonly'],
+    }, () => {
+
+        test('GET repositories', async ({request}) => {
+            const response = await request.get(url + "/repositories",
+                {
+                    headers: {
+                        authorization: 'Basic YWRtaW46YWRtaW4='
```

Comment:
Can you factorize this variable in `auth.setup.ts` (and same in `requests-metadata.spec.js`) ?